source/models/table.py

```diff
@@ -47,22 +53,65 @@ def set(self, row):
             old_row.delete()
         row_obj.serialize()
 
-    
-    def get(self):
-        pass
-    
+
+    def get_similar_primarykeys(self, query):
+        primarykeys = []
+        for index_name in query:
+            if index_name in self.table_metadata.index_keys:
+                primarykeys = Index(self, index_name = index_name, index_value = query[index_name]).get_primary_key(index_value = query[index_name])
+                break
```

Comment:
You can query with some parameters that aren't indexable, You assumed every key in the query is indexed, which is wrong. 
You should break if 
`self.table_metadata.indices[index_name] != null`

Reply:
yes i now that , if the query doesn't indexed and doesn't have the primarykey I take all the Data in the table then filter it

Reply:
please , check get function in table class again